hollow/src/main/java/com/netflix/hollow/core/write/objectmapper/HollowObjectMapper.java

```diff
@@ -91,6 +93,16 @@ public void writeFlat(Object o, FlatRecordWriter flatRecordWriter) {
     	typeMapper.writeFlat(o, flatRecordWriter);
     }
 
+    public <T> T readFlat(FlatRecordTraversalObjectNode node) {
```

Comment:
ask: let's keep this as the generic `FlatRecordTraversalNode` so that it can deser any of the four types. i think only the function parameter type changes, nothing else.
```suggestion
    public <T> T readFlat(FlatRecordTraversalNode node) {
```